xarray/tests/test_plot.py

```diff
@@ -2381,3 +2382,61 @@ def test_facetgrid_single_contour():
     ds["time"] = [0, 1]
 
     ds.plot.contour(col="time", levels=[4], colors=["k"])
+
+
+@pytest.mark.parametrize("figsize", [[4, 4], None])
+@pytest.mark.parametrize("aspect", [4 / 3, None])
+@pytest.mark.parametrize("size", [200, None])
+def test_get_axis(figsize, aspect, size):
+    # test get_axis works with different args combinations
+    # and return the right type
+    try:
+        import matplotlib.pyplot as plt
+    except ImportError:
+        with pytest.raises(ImportError):
+            ax = get_axis(figsize, size, aspect, None)
+        print("matplotlib not available")
+        return None
+
+    ax = "something"
+    if figsize is not None:
+        # cannot provide both ax and figsize
+        with pytest.raises(ValueError):
+            ax = get_axis(figsize, size, aspect, ax)
+    if size is not None:
+        # cannot provide both ax and size
+        with pytest.raises(ValueError):
+            ax = get_axis(figsize, size, aspect, ax)
+
+    ax = None
+    if figsize is not None and size is not None:
+        # cannot provide both size and figsize
+        with pytest.raises(ValueError):
+            ax = get_axis(figsize, size, aspect, ax)
+
+    if figsize is None:
+        if aspect is not None and size is None:
```

Comment:
Unless you're planning to add more code for the case where `figsize is None`, can you combine these conditions?